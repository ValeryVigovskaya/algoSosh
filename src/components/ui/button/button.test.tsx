import renderer from "react-test-renderer";
import { render, screen, fireEvent } from "@testing-library/react";
import { Button } from "./button";

describe("Testing the Button component", () => {

  it("should be the correct display of the button with text", () => {
    const tree = renderer.create(<Button text="Teст" />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be the correct display of the button without text", () => {
    const tree = renderer.create(<Button />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be the correct display of the locked button", () => {
    const tree = renderer.create(<Button disabled={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be the correct button with a loading indication", () => {
    const tree = renderer.create(<Button isLoader={true} />).toJSON();
    expect(tree).toMatchSnapshot();
  });

  it("should be a correct callback call when the button is clicked", () => {
    const callback = jest.fn();
    render(<Button onClick={callback} />);
    // Находим кнопку
    const button = screen.getByRole("button");
    // Имитируем нажатие на кнопку
    fireEvent.click(button);
    // Проверяем, что колбек был вызван
    expect(callback).toHaveBeenCalled();
  });
});
