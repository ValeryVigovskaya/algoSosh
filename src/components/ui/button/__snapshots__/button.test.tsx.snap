// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing the Button component should be the correct button with a loading indication 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Testing the Button component should be the correct display of the button with text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Teст
  </p>
</button>
`;

exports[`Testing the Button component should be the correct display of the button without text 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Testing the Button component should be the correct display of the locked button 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка без текста без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка заблокирована без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Кнопка с индикацией загрузки без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Кнопка с текстом без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Teст
  </p>
</button>
`;

exports[`Тестирование компонента Button Кнопка без текста без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка заблокирована без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка с индикацией загрузки без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Button Кнопка с текстом без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Teст
  </p>
</button>
`;

exports[`Тестирование компонента Circle Кнопка без текста без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Circle Кнопка заблокирована без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  disabled={true}
  type="button"
>
  <p
    className="text undefined"
  />
</button>
`;

exports[`Тестирование компонента Circle Кнопка с индикацией загрузки без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined loader "
  disabled={true}
  type="button"
>
  <img
    alt="Загрузка."
    className="loader_icon"
    src="loader.svg"
  />
</button>
`;

exports[`Тестирование компонента Circle Кнопка с текстом без ошибок 1`] = `
<button
  className="text text_type_button text_color_primary button undefined false "
  type="button"
>
  <p
    className="text undefined"
  >
    Teст
  </p>
</button>
`;
